lib/galaxy/jobs/runners/pulsar.py

```diff
@@ -265,6 +265,10 @@ def check_watched_item(self, job_state):
         if self.use_mq:
             # Might still need to check pod IPs.
             job_wrapper = job_state.job_wrapper
+            persisted_state = job_wrapper.get_state()
+            if persisted_state in model.Job.terminal_states + [model.Job.states.DELETED_NEW]:
+                log.debug("(%s) Watched job in terminal state, will stop monitoring: %s", job_state.job_id, persisted_state)
+                return None
             guest_ports = job_wrapper.guest_ports
```

Comment:
Probably should verify there are ports we want to check before fetching new state from the database?

Reply:
If there aren't then there's no need to monitor at all, correct?

Reply:
I think so. I guess there will be after we merge your other PR though.